sklearn/linear_model/_ridge.py

```diff
@@ -630,7 +630,12 @@ class Ridge(MultiOutputMixin, RegressorMixin, _BaseRidge):
     fit_intercept : bool, default=True
         Whether to fit the intercept for this model. If set
         to false, no intercept will be used in calculations
-        (i.e. ``X`` and ``y`` are expected to be centered).
+        (i.e. ``X`` and ``y`` are expected to be centered)
+        ``coef_`` returns same value irrespective of y is centered.
+        The model will underfit if ``fit_intercept=False``
+        and ``y`` is not centered.
```

Comment:
aren't these two sentences in contradiction?

If y-centering doesn't affect coefficients, then how y-centering can affect underfitting?

Reply:
As @thomasjpfan suggested, I am trying to mention the importance of centered y. When I ran the code and outputted the coef_ the results were the same with y_centered and y_uncentered.  But the mse with centered y is low. How should we frame a one-liner explanation for this?

Reply:
IIUC, what's causing confusion in the original issue is that when X is centered, centering y has no effect on the computation of the coefficients.

But that's unrelated to the fit_intercept parameter, so I'm not sure what needs to be done. The current description 

>          (i.e. ``X`` and ``y`` are expected to be centered) 

is correct: if you don't center y, you'll have a bad mse. It seems that the original issue author was assuming the models were equivalent just by looking at the coeff, but this is not a correct assumption since they didn't take the intercept into account.

Reply:
If we consider the original issue an usage issue, then we do nothing. (In this case, "usage issue" means "misinterpreted the model")

Reply:
I would agree with that yes.

I was myself surprised about this, but this isn't related to `fit_intercept` and I don't really see how we could discuss this fact without adding confusion.

(It comes from the fact that inv(Xt . X) . Xt . C is always equal to 0, when C is a constant vector.)

Reply:
I agree with that, it is not related to fit_intercept but and centered y vs not centered y is causing the confusion. 
Should we just write this or it's obvious to mention?
> (i.e. ``X`` and ``y`` are expected to be centered. Centered y gives lower ``mse``)

Reply:
> Centered y gives lower mse

This is only true for fit_intercept=False though ?

Reply:
I got nearly the same ``mse`` for both the cases.

1. Centered `y` when `fit_intercept=False` : `0.07086802779966515`

2.  Centered `y` when `fit_intercept=True` : 0.07086802779965488

Reply:
Hi, I wanted to ask what we have decided to do here? Do you think we should make any changes to the DOC?

Reply:
Thank you for working on this @yashika51 

Let's close this PR, I think the docstring is clear enough. I think adding more seems to make it more confusing.